### PR TITLE
Kraigo notifications

### PR DESCRIPTION
**Pull request by myConsciousness**
# 1. Description

## 1.1. Checklist

- [x] The title of my PR starts with a [Conventional Commit] prefix (`fix:`, `feat:`, `docs:` etc).
- [x] I have read the [Contributor Guide] and followed the process outlined for submitting PRs.
- [x] I have updated/added tests for ALL new/updated/fixed functionality.
- [x] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [ ] I have updated/added relevant examples in `examples`.

## 1.2. Breaking Change

- [ ] Yes, this is a breaking change.
- [x] No, this is _not_ a breaking change.

## 1.3. Related Issues

[issue database]: https://github.com/mastodon-dart/mastodon-api/issues
[contributor guide]: https://github.com/mastodon-dart/mastodon-api/blob/main/CONTRIBUTING.md
[style guide]: https://github.com/mastodon-dart/mastodon-api/blob/main/STYLEGUIDE.md
[conventional commit]: https://conventionalcommits.org

### CODE REVIEW

**Comment by myConsciousness** [comment]
bors merge

**Comment by myConsciousness** [comment]
Hi @Kraigo ,

Thanks for your contributions! These commits will be released as v0.4.0. :)

File: test/src/service/v1/notifications/notifications_v1_service_test.dart [hunk 1]
```diff
@@ -49,70 +50,218 @@ void main() {
...
+      );
+
+      expectUnauthorizedException(
+        () async => await notificationsService.destroyNotification(
+          notificationId: '12345',
+        ),
+      );
+    });
+
+    test('when rate limit exceeded', () async {
```
Comment (Kraigo) [inline test/src/service/v1/notifications/notifications_v1_service_test.dart:248]: Hey, @myConsciousness why do you need every time test rate limit and authorization response for each request?
I see that is stubbed data which comes from BaseService and all method use that solution.
An endpoint methods should test for available parameters and expected response object.
Did I miss something? What cases can such a case cover?

Comment (myConsciousness) [reply test/src/service/v1/notifications/notifications_v1_service_test.dart:248]: Hi @Kraigo ,

These are tests to ensure that each endpoint can reliably handle exception cases :) An exception case that endpoint A can handle perfectly and endpoint B returns a different exception or error would create chaos.

The processing in this package is commonized like a framework, but it’s up to the implementer to be able to use and implement it correctly. So, even if the code is redundant, these common specifications must be tested to ensure their correctness :)

**Pull request merged**